### PR TITLE
Add Win32-compatible version of "popcount" and "bitscan"

### PR DESCRIPTION
**Pull request by DanielUranga**
Add win32-compatible versions of those functions so it doesn't fail to compile when using 32 bits Visual Studio.

### CODE REVIEW

File: src/utils/bititer.h [hunk 1]
```diff
@@ -44,9 +44,20 @@ class BitIterator {
   void operator++() { value_ &= (value_ - 1); }
   T operator*() const {
 #ifdef _MSC_VER
+#ifdef _WIN64
     unsigned long result;
     _BitScanForward64(&result, value_);
     return result;
+#else
+    unsigned long result;
+    if (result & 0xFFFFFFFF) {
```
Comment (borg323) [inline src/utils/bititer.h]: This should be `value_ & 0xFFFFFFFF`

Comment (DanielUranga) [reply src/utils/bititer.h]: Whoops!

**Comment by mooskagh** [comment]
Not commenting anything else, it's better to have
```cpp
#elif defined(BLAH)
```
instead of nested `#else #ifdef BLAH #endif #endif`s

File: src/chess/bitboard.h [hunk 2]
```diff
@@ -97,7 +97,11 @@ class BitBoard {
   void clear() { board_ = 0; }
   int count() const {
 #ifdef _MSC_VER
+#ifdef _WIN64
```
Comment (borg323) [inline src/chess/bitboard.h]: Can you rearrange it to avoid nested #ifdef blocks?

File: src/utils/bititer.h [hunk 3]
```diff
@@ -44,9 +44,20 @@ class BitIterator {
   void operator++() { value_ &= (value_ - 1); }
   T operator*() const {
 #ifdef _MSC_VER
+#ifdef _WIN64
```
Comment (borg323) [inline src/utils/bititer.h]: Here as well

**Comment by DanielUranga** [comment]
I see this is approved, but I can't merge. For some reason all of my PRs always have CircleCI in "pending" status.

**Pull request merged**